### PR TITLE
feat: filter parameters from DB API

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

Filter out some of the parameters from the database connection in the API.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by AAfghahi — APPROVED** [review]

File: superset/models/core.py [hunk 1]
```diff
@@ -339,14 +339,6 @@ def get_effective_user(self, object_url: URL) -> Optional[str]:
             else None
         )
 
-    @memoized(
```
Comment (betodealmeida) [inline superset/models/core.py:342]: This is not working for BigQuery, had to disable.

File: superset/models/core.py [hunk 2]
```diff
@@ -639,7 +631,6 @@ def db_engine_spec(self) -> Type[db_engine_specs.BaseEngineSpec]:
         return self.get_db_engine_spec(url)
 
     @classmethod
-    @memoized
```
Comment (betodealmeida) [inline superset/models/core.py:642]: Same here, not working for BigQuery.

**Review by eschutho — APPROVED** [review]

**Review by villebro — APPROVED** [review]
LGTM

File: superset/db_engine_specs/bigquery.py [hunk 3]
```diff
@@ -380,18 +381,47 @@ def build_sqlalchemy_uri(
...
+                pass
+
             # ``value.query`` needs to be explicitly converted into a dict (from an
             # ``immutabledict``) so that it can be JSON serialized
             return {**encrypted_extra, "query": dict(value.query)}
 
         raise ValidationError("Invalid service credentials")
 
+    @classmethod
+    def update_encrypted_extra(
```
Comment (dpgaspar) [inline superset/db_engine_specs/bigquery.py:404]: What do you think about adding a unit test to bigquery and gsheets. An actual test for this has an integration test for the API would be nice also

**Review by dpgaspar — APPROVED** [review]

**Review by dpgaspar — COMMENTED** [review]
tests are failing

File: superset/db_engine_specs/bigquery.py (hunk 3 above)
Comment (betodealmeida) [reply superset/db_engine_specs/bigquery.py:404]: Working on it!

File: superset/databases/dao.py [hunk 4]
```diff
@@ -33,6 +34,25 @@ class DatabaseDAO(BaseDAO):
...
 
+    @classmethod
+    def update(
+        cls,
+        model: Database,
+        properties: Dict[str, Any],
+        commit: bool = True,
+    ) -> Database:
+        if "encrypted_extra" in properties:
+            new = json.loads(properties["encrypted_extra"])
```
Comment (hughhhh) [inline superset/databases/dao.py:45]: shouldn't we have try/except here just incase we have malformed coming in from the modal?

Comment (betodealmeida) [reply superset/databases/dao.py:45]: Yeah, good point, that's safer. Let me add.

**Comment by jinghua-qa** [comment]
for the skip test, do we have a plan to get it fix and re-enable?

**Comment by betodealmeida** [comment]
> for the skip test, do we have a plan to get it fix and re-enable?

I'm not sure why these were timing out, they are unrelated to this PR. I need to take a better look.

**Review by jinghua-qa — APPROVED** [review]
LGTM

**Comment by sadpandajoe** [comment]
🏷️ preset:2022.33

**Pull request merged**